### ISSUE TITLE
Q: equivalent to old CLI syntax?

### ISSUE THREAD

**Issue by sloanlance** [python-jsonschema/check-jsonschema#328]
In the (long distant) past, I had used the `jsonschema` CLI included in [python-jsonschema/jsonschema](/python-jsonschema/jsonschema).  I'm now transitioning to `check-jsonschema`, but I can't find an equivalent to the syntax I used with the old tool.

I used `jsonschema` while I'm writing JSON schemas to let me test a small piece of schema syntax with some input.  A contrived example of the syntax I had used…

```sh
$ printf '7' | jsonschema <(printf '{"minimum": 9, "multipleOf": 3, "not": {"const": 7}}')
jsonschema:5: DeprecationWarning: The jsonschema CLI is deprecated and will be removed in a future version. Please use check-jsonschema instead, which can be installed from https://pypi.org/project/check-jsonschema/
  from jsonschema.cli import main
7: 7 is less than the minimum of 9
7: 7 is not a multiple of 3
7: 7 should not be valid under {'const': 7}
```

I.e., it tests the input (`7`) against the schema, `{"minimum": 9, "multipleOf": 3, "not": {"const": 7}}`.  As shown here, `jsonschema` continues to work, but I get the deprecation warning.

If I directly replace `jsonschema` with `check-jsonschema` in the command, it fails…

```sh
$ printf '7' | check-jsonschema <(printf '{"minimum": 9, "multipleOf": 3, "not": {"const": 7}}')
Usage: check-jsonschema [OPTIONS] INSTANCEFILES...

Error: Either --schemafile, --builtin-schema, or --check-metaschema must be provided
```
I've tried using the `--schemafile` option, but haven't found an argument to it (e.g., `-` or `/dev/stdin`) that tells the tool the schema is provided in the input or arguments.

I've not figured out the correct alternate syntax I should use.  Could anyone suggest one to me?

Thanks in advance!

**Comment by sirosen** [comment]
So, I'll note right off the bat that I made some decisions in the initial CLI interface for `check-jsonschema` which I'd like to take back. Being more compatible with the older built-in `jsonschema` CLI out of the box would be a target for me if I were doing it all over again.

The syntax I would use for your case today is two subshells:
```
check-jsonschema --schemafile <(printf '{"minimum": 9, "multipleOf": 3, "not": {"const": 7}}') <(printf '7')
```
which when I ran it just now got me...
```
$ check-jsonschema --schemafile <(printf '{"minimum": 9, "multipleOf": 3, "not": {"const": 7}}') <(printf '7')
Schema validation errors were encountered.
  /proc/self/fd/17::$: 7 is less than the minimum of 9
  /proc/self/fd/17::$: 7 is not a multiple of 3
  /proc/self/fd/17::$: 7 should not be valid under {'const': 7}
```

Two quick notes for compatibility and porting your usages:
- `check-jsonschema` requires `--schemafile SCHEMA` whereas `jsonschema` took it as the first positional arg
- `check-jsonschema` *currently* doesn't support passing `-` to indicate stdin, but I consider that a missing feature and have it on my roadmap to fix/add it

Not supporting stdin from day 1 has unfortunately made some of the internals messy.
